### DISCUSSION TITLE
How to add pattern for date

### DISCUSSION THREAD

**Discussion by sandeepsre** [mantinedev/mantine#1984] [Q&A]
Hi there,
I am using TextInput for Date in case of date format (MM/DD/YYYY). How I can use HTML 5 Pattern for date.
I am using below expression
pattern="(?:19|20)[0-9]{2}-(?:(?:0[1-9]|1[0-2])-(?:0[1-9]|1[0-9]|2[0-9])|(?:(?!02)(?:0[1-9]|1[0-2])-(?:30))|(?:(?:0[13578]|1[02])-31))"

**Comment by rtivital** [comment]
You probably should use react-input-mask with InputBase component instead of TextInput – https://mantine.dev/core/input/#inputbase-component

**Reply by sandeepsre** [reply]
InputBase is not working?
![image](https://user-images.githubusercontent.com/16818395/182618369-9fb0e334-a137-42ce-8ea8-2b4d7ca8573b.png)

**Reply by rtivital** [reply]
You need to update to 5.0.2 to use it. For versions 4.x you can use `Input` component instead.

**Reply by sandeepsre** [reply]
How i can Update  versions 4.x to 5.0.2

**Reply by rtivital** [reply]
Change versions in package.json and install dependencies

**Reply by sandeepsre** [reply]
"dependencies": {
    "@mantine/core": "^4.2.2",
    "@mantine/dates": "^4.2.5",
    "@mantine/dropzone": "^4.2.5",
    "@mantine/hooks": "^4.2.2",
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^13.2.0",
    "@testing-library/user-event": "^13.5.0",
    "axios": "^0.27.2",
    "formik": "^2.2.9",
    "history": "^5.3.0",
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    "react-dropzone": "^14.2.2",
    "react-icons": "^4.3.1",
    "react-input-mask": "^2.0.4",
    "react-player": "^2.10.1",
    "react-router-dom": "^6.3.0",
    "react-scripts": "5.0.1",
    "serve": "^14.0.1",
    "tabler-icons-react": "^1.54.0",
    "validator": "^13.7.0",
    "web-vitals": "^2.1.4"
  },

need to change first 4, or only first?
